src/scripts/outlet-registration/index.js

```diff
@@ -0,0 +1,582 @@
+    //Prepare orgUnit
+    var newOrgUnit = this.createOrgUnitFromEvent(event);
+    //Post orgunit  
+    if (!this.ifValidOrgUnit(newOrgUnit)) {
+        console.log("Compulsory fields have not been filled ",newOrgUnit);
+        this.nextEvent();
+        return
+    }
+    this.postAndPatch(newOrgUnit, event);
+  
+};
+
+/**
+ * Returns a string with outlet code
+ * {AMTR}{ParentCode}{-}{Increment}
+ * @params parentCode of the parent org. unit
+ * @params autoIncrement integer to be added to the code
+ */
+OutletRegistrator.prototype.createOrgUnitCode = function(parentCode,autoIncrement) {
+    var formatNumber = (autoIncrement>0 && autoIncrement<10)?"0"+autoIncrement:autoIncrement;
+    return "AMTR"+parentCode+"-"+formatNumber;
+};
+
+/**
+ * DHIS2 format for coordinates [longitude,latitude]
+ * @params coord [longitude, latitude] coordinates
+ */
+OutletRegistrator.prototype.setupCoordiantes = function(coord) {
+    coordinates = [coord.longitude, coord.latitude];
+    return coordinates;
+};
+
+/**
+ * Returns an orgUnit with every required field
+ * @param event The event with the data
+ */
+OutletRegistrator.prototype.createOrgUnitFromEvent = function(event) {
+
+    var newOu = {};
+    //get outlet name
+    var outletName = this.findDataValue(event, this.conf.dataElements.name);
+    //get outlet code
+    var outletCode = this.createOrgUnitCode(event.parentCode, event.autoIncrement);
+    //get outlet contact person
+    var outletContactPerson = this.findDataValue(event, this.conf.dataElements.contactPerson);	
+    //get outlet address
+    var outletAddress = this.findDataValue(event, this.conf.dataElements.address);
+    //get outlet phone number
+    var outletPhoneNumber = this.findDataValue(event, this.conf.dataElements.phoneNumber);
+    //get outlet type
+    var outletType = this.findDataValue(event, this.conf.dataElements.outletType);
+    
+    newOu.code=this.myanmarPrefix+outletCode;
+    newOu.name=outletName + " (" + outletCode + ")";
+    newOu.shortName=outletName;
+    newOu.openingDate=event.eventDate.split('T')[0]; //Removing the time
+    newOu.featureType="POINT";
+    newOu.parent={
+        id:event.orgUnit
+    };
+    newOu.address=outletAddress;
+    newOu.phoneNumber=outletPhoneNumber;
+    newOu.contactPerson=outletContactPerson;
+    newOu.outletType=outletType;
+    //(0,0) means the coordinates have not been pushed from android
+    if (event.coordinate.longitude!=0 ||  event.coordinate.latitude!=0)
+    	newOu.coordinates=JSON.stringify(this.setupCoordiantes(event.coordinate));
+    
+    return newOu;
+}
+
+/**
+ * Get the user from the event.
+ * We need to get the user from the storedBy attribute of the datavalues
+ * @params event
+ */
+OutletRegistrator.prototype.getUserFromEvent = function(event) {
+    var dataValues = event.dataValues;
+    var username="";
+    
+    if (dataValues.length>0){
+    	username = event.dataValues[0].storedBy;
```

Comment:
Either delete dataValues variables and use event.dataValues or use dataValues everywhere. In general, I would simplify the whole function with just a ternary